tests/test_database_types.py

```diff
@@ -1,30 +1,88 @@
 import unittest
 import preql
 import time
+import datetime
 from data_diff import database as db
 from data_diff.diff_tables import TableDiffer, TableSegment, split_space
 from parameterized import parameterized, parameterized_class
-from .common import CONN_STRINGS, str_to_checksum
+from .common import CONN_STRINGS, DEFAULT_N_SAMPLES, N_SAMPLES, str_to_checksum
 import logging
 
-logging.getLogger("diff_tables").setLevel(logging.WARN)
-logging.getLogger("database").setLevel(logging.WARN)
-
 CONNS = {k: db.connect_to_uri(v) for k, v in CONN_STRINGS.items()}
-
 CONNS[db.MySQL].query("SET @@session.time_zone='+00:00'", None)
 
+
+class Faker:
+    pass
+
+
+class DateTimeFaker:
+    def __init__(self, max):
+        self.max = max
+
+    def __iter__(self):
+        self.prev = datetime.datetime(2000, 1, 1, 0, 0, 0, 0)
+        self.i = 0
+        return self
+
+    def __len__(self):
+        return self.max
+
+    def __next__(self) -> str:
+        if self.i < self.max:
+            self.prev = self.prev + datetime.timedelta(seconds=3, microseconds=571)
+            self.i += 1
+            return str(self.prev)
+        else:
+            raise StopIteration
+
+
+class IntFaker:
+    def __init__(self, max):
+        self.max = max
+
+    def __iter__(self):
+        self.prev = -128
+        self.i = 0
+        return self
+
+    def __len__(self):
+        return self.max
+
+    def __next__(self) -> str:
+        if self.i < self.max:
+            self.prev += 1
+            self.i += 1
+            return str(self.prev)
+        else:
+            raise StopIteration
+
+
+class FloatFaker:
+    def __init__(self, max):
+        self.max = max
+
+    def __iter__(self):
+        self.prev = -10.0001
+        self.i = 0
+        return self
+
+    def __len__(self):
+        return self.max
+
+    def __next__(self) -> str:
+        if self.i < self.max:
+            self.prev += 0.00571
+            self.i += 1
+            return str(self.prev)
+        else:
+            raise StopIteration
+
+
 TYPE_SAMPLES = {
-    "int": [127, -3, -9, 37, 15, 127],
-    "datetime_no_timezone": [
-        "2020-01-01 15:10:10",
-        "2020-01-01 9:9:9",
-        "2022-01-01 15:10:01.139",
-        "2022-01-01 15:10:02.020409",
-        "2022-01-01 15:10:03.003030",
-        "2022-01-01 15:10:05.009900",
-    ],
-    "float": [0.0, 0.1, 0.10, 10.0, 100.98],
+    "int": IntFaker(N_SAMPLES),
```

Comment:
This feels like a step backwards. How can we be sure that the "faker" creates all the edge cases that we are interested in?

Reply:
I think it generates plenty of edge-cases naturally by creating lots of 9s, 8s, etc.

```
['2000-01-01 00:00:03.000571', '2000-01-01 00:00:06.001142', '2000-01-01 00:00:09.001713', '2000-01-01 00:00:12.002284', '2000-01-01 00:00:15.002855', '2000-01-01 00:00:18.003426', '2000-01-01 00:00:21.003997', '2000-01-01 00:00:24.004568', '2000-01-01 00:00:27.005139', '2000-01-01 00:00:30.005710', '2000-01-01 00:00:33.006281', '2000-01-01 00:00:36.006852', '2000-01-01 00:00:39.007423', '2000-01-01 00:00:42.007994', '2000-01-01 00:00:45.008565', '2000-01-01 00:00:48.009136', '2000-01-01 00:00:51.009707', '2000-01-01 00:00:54.010278', '2000-01-01 00:00:57.010849', '2000-01-01 00:01:00.011420', '2000-01-01 00:01:03.011991', '2000-01-01 00:01:06.012562', '2000-01-01 00:01:09.013133', '2000-01-01 00:01:12.013704', '2000-01-01 00:01:15.014275', '2000-01-01 00:01:18.014846', '2000-01-01 00:01:21.015417', '2000-01-01 00:01:24.015988', '2000-01-01 00:01:27.016559', '2000-01-01 00:01:30.017130', '2000-01-01 00:01:33.017701', '2000-01-01 00:01:36.018272', '2000-01-01 00:01:39.018843', '2000-01-01 00:01:42.019414', '2000-01-01 00:01:45.019985', '2000-01-01 00:01:48.020556', '2000-01-01 00:01:51.021127', '2000-01-01 00:01:54.021698', '2000-01-01 00:01:57.022269', '2000-01-01 00:02:00.022840', '2000-01-01 00:02:03.023411', '2000-01-01 00:02:06.023982', '2000-01-01 00:02:09.024553', '2000-01-01 00:02:12.025124', '2000-01-01 00:02:15.025695', '2000-01-01 00:02:18.026266', '2000-01-01 00:02:21.026837', '2000-01-01 00:02:24.027408', '2000-01-01 00:02:27.027979', '2000-01-01 00:02:30.028550']
```

Reply:
But there are edge-cases we already know are problematic. It doesn't make sense to throw them away, and hope we get them through an equation..

Reply:
Ok. I can hard-code the first N of the faker to the existing hard-coded values

Reply:
That's acceptable.